### PR TITLE
1793: Reports: AbstractDialog Swapover, GUI Isolation, Localization Preference Bugfixes, and Code Standardization

### LINKED ISSUE #1793: Command Center: Add Preferences to Hangar Breakdown Report
This RFE calls for adding dialog preference to the Hangar Breakdown Report Dialog, as you currently need to expand it significantly to use it in most cases every time you load the dialog.

### PR DESCRIPTION
**Pull request by Windchild292**
This is a rework of the general report setup (only making changes to the CargoReport), and a complete rewrite of the dialogs.
It replaces #2814 and #2815, which gave me the overall rework idea.

This fixes #1793

### CODE REVIEW

File: MekHQ/src/mekhq/campaign/unit/Unit.java [hunk 1]
```diff
@@ -129,7 +129,7 @@
     private Campaign campaign;
 
     private ArrayList<Part> parts;
-    private String lastMaintenanceReport;
```
Comment (Windchild292) [inline MekHQ/src/mekhq/campaign/unit/Unit.java:132]: Remove TODO

File: MekHQ/src/mekhq/gui/dialog/reportDialogs/AbstractReportDialog.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import mekhq.gui.baseComponents.AbstractMHQDialog;
+
+import javax.swing.*;
+import javax.swing.border.EmptyBorder;
+import java.awt.*;
+
+/**
+ * This displays a standard report for MekHQ.
+ *
+ * Inheriting classes must call initialize() in their constructors and override getText
```
Comment (Windchild292) [inline MekHQ/src/mekhq/gui/dialog/reportDialogs/AbstractReportDialog.java:30]:
```suggestion
 * Inheriting classes must call initialize() in their constructors and override getText().
```

File: MekHQ/src/mekhq/gui/dialog/reportDialogs/PersonnelReportDialog.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        return personnelReport;
+    }
+
+    @Override
+    protected JTextPane createTxtReport() {
+        final JTextPane txtCombatPersonnel = new JTextPane();
+        txtCombatPersonnel.setText(getPersonnelReport().getCombatPersonnelDetails());
+
+        final JTextPane txtSupportPersonnel = new JTextPane();
+        txtSupportPersonnel.setText(getPersonnelReport().getSupportPersonnelDetails());
```
Comment (Windchild292) [inline MekHQ/src/mekhq/gui/dialog/reportDialogs/PersonnelReportDialog.java:49]: Both are missing .setName

**Review by Windchild292 — COMMENTED** [review]
Some personal review notes

**Review by NickAragua — APPROVED** [review]

**Pull request merged**